api/src/components/reference/controller.ts

```diff
@@ -14,7 +15,29 @@ export const get = async (event: I.AuthenticatedAPIRequest<undefined, undefined,
 
 export const create = async (event: I.AuthenticatedAPIRequest<I.Reference, undefined, I.CreateReferencePath>) => {
     try {
-        const reference = await referenceService.create({ publicationId: event.pathParameters.id, ...event.body });
+        const publication = await publicationService.get(event.pathParameters.id);
+
+        //check that the publication exists
+        if (!publication) {
+            return response.json(404, {
+                message: 'This publication does not exist.'
+            });
+        }
+
+        //check that the publication is live
+        if (publication.currentStatus !== 'DRAFT') {
+            return response.json(403, {
+                message: 'You can only add references to a draft publication.'
+            });
+        }
+
+        if (event.body.publicationId !== event.pathParameters.id || event.body.id == '' || event.body.text == '') {
```

Comment:
ok this is part of what i created OCT-347 for, but both update and updateall need to do the same checks